enumerables.rb

```diff
@@ -0,0 +1,131 @@
+module Enumerable
+  def my_each
+    size = count - 1
+    counter = 0
+    items = to_a
+    while counter <= size
+      if is_a? Range
+        items.length >= 0 && yield(items[counter])
+      elsif is_a? Hash
+        yield(items[counter][0], items[counter][1])
+      elsif !block_given?
+        return self
+      else
+        yield(self[counter])
+      end
+      counter += 1
+    end
+    self
+  end
+
+  def my_each_with_index()
+    counter = 0
+    size = length - 1
+    return self unless block_given?
+
+    while counter <= size
+      yield(self[counter], counter)
+      counter += 1
+    end
+    self
+  end
+
+  def my_select
+    return self unless block_given?
+
+    required = []
+
+    my_each do |r|
+      required.push(yield(r))
+    end
+    required
+  end
+
+  def my_all?(*)
+    return self unless block_given?
+
+    track_negative = 0
+
+    my_each do |t|
+      yield(t) == true ? true : track_negative += 1
+    end
+    !track_negative.positive?
+  end
```

Comment:
#### It is important to note that `my_any?`, `my_all?` and `my_none?` will always return `true` or `false`

 - When a pattern is passed this method should return true if every collection member matches this pattern. Note that pattern could be a `Class` (e.g. `Integer`, `String`) or `RegEx`
   ```ruby
   # When a Class pattern is passed
   irb(main):014:0> [1,2,3].all?(Integer)
   => true
   irb(main):015:0> [1,2,3].my_all?(Integer)
   => false

   # When a RegEx pattern is passed
   irb(main):018:0> %w[ant bat cat].all?(/t/)
   => true
   irb(main):019:0> %w[ant bat cat].my_all?(/t/)
   => false
   ```
   Hint: Use ` === ` to check if a member matches a patern.

### Find more details about [`all?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F)

### Please let's do this for `my_any?` and `my_none?` too.